tests/rptest/tests/full_disk_test.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2022 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+import random
+from time import sleep, time
+
+from ducktape.cluster.cluster import ClusterNode
+from ducktape.tests.test import TestContext
+from ducktape.utils.util import wait_until
+from rptest.clients.types import TopicSpec
+from rptest.services.admin import Admin
+from rptest.services.cluster import cluster
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.utils.node_metrics import NodeMetrics
+from rptest.util import wait_until_result
+
+from kafka import KafkaProducer
+
+N = 3
+# reduce this?
+MAX_MSG = 500
+MAX_MSG_PER_SEC = 10
+FDT_LOG_ALLOW_LIST = [".*cluster - storage space alert: free space.*"]
+
+
+class FullDiskTest(RedpandaTest):
+
+    TOPIC = "fulldisktest-topic"
+    CONF_MIN_FREE_BYTES = "storage_min_free_bytes"
+
+    def __init__(self, test_context: TestContext):
+        self.topics = [TopicSpec(name=self.TOPIC)]
+        super().__init__(test_context)
+
+    # TODO factor out lib for this, common with cluster_config_test.py
+    def _wait_for_node_config_value(self, node: ClusterNode, key: str,
+                                    value: int) -> None:
+        _get = lambda k: self.admin.get_cluster_config(node=node)[k]
```

Comment:
i think you should be able to avoid passing in a specific node here and letting the library choose a random node

Reply:
Yep I already got this one--you'd mentioned it earlier. Thank you.